CIP-1694/README.md

```diff
@@ -0,0 +1,664 @@
+4. For updates to the Constitution - a 32-byte blake2b-256 hash digest of the Constitution document;
+5. For new Constitutional Committee and changes to the quorum size - a set of key hashes and a positive number that is no greater than the size of the committee;
+6. For a motion of no confidence - nothing else.
+
+The deposit amount will be added to the _deposit pot_, similar to stake key deposits.
+
+#### Governance Action IDs
+
+Each accepted governance action will be assigned a unique identifier (a.k.a. the **governance action ID**), consisting of the transaction ID that created it and the index within the transaction body that points to it.
+
+### Votes
+
+Each vote transaction consists of the following:
+
+* a governance action ID;
+* a role - Constitutional Committee, DRep, or SPO;
+* a key-hash (which will be verified to have the role above);
+* a URL for any metadata that is relevant to the vote;
+* a hash of the contents of this URL;
+* a yes/no/abstain vote.
+
+Note that "abstained" votes are not included in the "active voting stake".
+Note also that a vote to abstain is different than abstaining from voting
+(the former is visible on chain, the latter is not).
+To avoid confusion, we will only use the word "abstain" from this point onward to mean a vote
+to abstain.
+
+The key hash will trigger the appropriate signature check on the transaction body according to the existing `UTxOW` ledger rule.
+
+Votes can be cast multiple times for each governance action by a single key hash. Correctly submitted votes override any older votes for the same key hash and role.
+As soon as a governance action is ratified, voting ends. Any future votes are not considered or recorded (whether they are yes, no or abstain).
+
+#### Governance State
+
+When a governance action is successfully submitted to the chain, its progress will be tracked by the ledger state.
+In particular, the following will be tracked:
+
+* the governance action ID;
+* the epoch that the action expires;
+* the deposit amount;
+* the rewards address that will receive the deposit when it is returned;
+* the total yes/no/abstain votes of the Constitutional Committee for this action;
+* the total yes/no/abstain votes of the DReps for this action;
+* the total yes/no/abstain votes of the SPOs  for this action.
+
+#### Stale Votes
+
+The votes from the DReps and the SPOs may become meaningless after crossing an epoch boundary since they may become unregistered. Therefore, all unregistered votes are cancelled before any new votes are considered.
+
+#### Changes to the Stake Snapshot
+
+Since the stake snapshot changes at each epoch boundary, a new current voting tally must be calculated for each governance action based on the votes that have been cast before any new votes are counted.  This avoids the same stake being used two or more times.  _This new tally may result in immediate ratification of a governance action._
+
+### Delegated Representatives (DReps)
```

Comment:
As a safety mechanism, I think it would be a good idea to introduce a delegation saturation parameter similar to stake pools. This would limit the voting power of a DRep in proportion to their own stake, e.g. if you only hold 1 ADA, then your delegated stake would be limited to 100 ADA.

Reply:
That sounds very dangerous, since the big guys then want this to be as small as possible so that they have more power. It seems to me that having no limit on this actually protects the small guys.

Reply:
Yeah, you are right. I missed to consider that.